.github/workflows/release-doxygen.yml

```diff
@@ -0,0 +1,62 @@
+name: Release Doxygen
+
+permissions:
+  contents: read
+
+on:
+  workflow_dispatch:
+    inputs:
+      release-version:
+        description: 'Release Version'
+        required: true
+        type: string
+      upload:
+        description: 'Upload documentation'
+        required: false
+        type: boolean
+
+  workflow_call:
+    inputs:
+      release-version:
+        description: 'Release Version'
+        required: true
+        type: string
+      upload:
+        description: 'Upload documentation'
+        required: false
+        type: boolean
+
+jobs:
+  release-doxygen:
+    name: Build and Upload Release Doxygen
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    env:
+      upload: ${{ inputs.upload && !contains(inputs.release-version, 'rc') }}
+    steps:
+      - name: Checkout LLVM
+        uses: actions/checkout@v4
+
+      - name: Install Dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y \
+              doxygen \
```

Comment:
Isn't doxygen already listed as a requirement in the pip requirements file?

It might be worth adding the pip cache with `setup-python` since it makes the install so much faster.

Reply:
doxygen is not in the requirements file.